packages/frontend/src/ee/features/aiCopilot/components/ChatElements/AgentChatAssistantBubble.tsx

```diff
@@ -362,53 +432,16 @@ export const AssistantBubble: FC<Props> = memo(
                 />
 
                 {isVisualizationAvailable && (
-                    <Paper
-                        withBorder
-                        radius="md"
-                        p="md"
-                        h="500px"
-                        shadow="none"
-                        {...((queryExecutionHandle.isError ||
-                            queryExecutionHandle.isLoading) && {
-                            bg: 'gray.0',
-                            style: {
-                                borderStyle: 'dashed',
-                            },
-                        })}
-                    >
-                        {isQueryLoading ? (
-                            <Center h="100%">
-                                <Loader
-                                    type="dots"
-                                    color="gray"
-                                    delayedMessage="Loading visualization..."
-                                />
-                            </Center>
-                        ) : isQueryError ? (
-                            <Stack
-                                gap="xs"
-                                align="center"
-                                justify="center"
-                                h="100%"
-                            >
-                                <MantineIcon
-                                    icon={IconExclamationCircle}
-                                    color="gray"
-                                />
-                                <Text size="xs" c="dimmed" ta="center">
-                                    Something went wrong generating the
-                                    visualization, please try again
-                                </Text>
-                            </Stack>
-                        ) : (
-                            <AiChartVisualization
-                                results={queryResults}
-                                message={message}
-                                queryExecutionHandle={queryExecutionHandle}
-                                projectUuid={projectUuid}
-                            />
-                        )}
-                    </Paper>
+                    <AiArtifactButton
+                        onClick={() =>
+                            setArtifact(renderArtifact(), message.uuid)
```

Comment:
we will use artifact uuid once we have the backend in place, and we could move this logic to a new place, where we can make the queries from that component instead of passing everything here?

Reply:
yes sounds good!